client/src/components/messenger/Settings.js

```diff
@@ -1,5 +1,168 @@
+import edit_icon from "../../assets/icons8-edit.svg"
+
+import { useState } from 'react';
+
+// this is just test data
+// this data should really be fetched
+// and then posted when updated and validated
+let user_data = {
+    display_name: "Cassie",
+    first_name: "Cassandra",
+    last_name: "Stevens",
+    image_url: "https://picsum.photos/200"
+}
+
 const Settings = () => {
-    return <h1>Settings</h1>
+    const [Name, setName] = useState({
+        firstname: user_data.first_name,
+        lastname: user_data.last_name,
+        editing: false, // determines whether name is being edited
+        editLock: false,
+    });
+
+    const handleNameChange = (e) => {
+        const { name, value } = e.target
+        Name[name] = value;
+        // setName({ ...Name, [name]: value })
+    }
+
+    const [displayName, setDisplayName] = useState({
+        displayname: user_data.display_name,
+        editing: false, // determines whether displayname is being edited
+        editLock: false, // locks if user has entered invalid data
+    });
+
+    const handleDNameChange = (e) => {
+        const { name, value } = e.target
+        setDisplayName({ ...displayName, [name]: value })
+        console.log(displayName);
+    }
+
+    const handleEditName = () => {
```

Comment:
I'm no react expert but this looks like manual DOM manipulation. Is this the right way of doing things?